File: VideoCallSDK/app/src/main/java/com/vnptit/videocallsample/db/ContactRepository.java
package com.vnptit.videocallsample.db;

import android.app.Application;

import androidx.lifecycle.LiveData;

import com.vnptit.videocallsample.model.Contact;

import java.util.List;

public class ContactRepository {
    private ContactDao contactDao;
    private LiveData<List<Contact>> mAllContacts;

    public ContactRepository(Application application) {
        AppDatabase db = AppDatabase.getDatabase(application);
        contactDao = db.contactDao();
        mAllContacts = contactDao.getAll();
    }

    public LiveData<List<Contact>> getAllContacts() {
        return mAllContacts;
    }

    public void insert(Contact contact) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            contactDao.insert(contact);
        });
    }

    public void update(Contact contact) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            contactDao.update(contact);
        });
    }

    public void delete(Contact contact) {
        AppDatabase.databaseWriteExecutor.execute(() -> {
            contactDao.delete(contact);
        });
    }
}

File: VideoCallSDK/kotlin/src/main/java/com/vnptit/videocallsample/db/ContactRepository.kt
package com.vnptit.videocallsample.db

import android.app.Application
import android.content.Context
import androidx.lifecycle.LiveData
import com.vnptit.videocallsample.model.ContactCall
import java.util.concurrent.Executors

/**
 * Created by Ông Hoàng Nhật Phương on 1/26/2021.
 * VNPT COM
 * phuonghn@vnpt.vn
 */
class ContactRepository(context: Context) {
    private var contactDao: ContactDao? = AppDataBase.getInstance(context)?.contactDao()
    private var mAllContacts: LiveData<List<ContactCall>>?

    private val databaseWriteExecutor = Executors.newFixedThreadPool(NUMBER_OF_THREADS)

    init {
        mAllContacts = contactDao?.getAll()
    }


    fun getAllContacts(): LiveData<List<ContactCall>>? {
        return mAllContacts
    }

    fun insert(contactCall: ContactCall) {
        databaseWriteExecutor.execute { contactDao?.insertContact(contactCall) }
    }

    fun update(contactCall: ContactCall) {
        databaseWriteExecutor.execute { contactDao?.update(contactCall) }
    }

    fun delete(contactCall: ContactCall) {
        databaseWriteExecutor.execute { contactDao?.delete(contactCall) }
    }

    companion object {
        private const val NUMBER_OF_THREADS = 2
    }
}